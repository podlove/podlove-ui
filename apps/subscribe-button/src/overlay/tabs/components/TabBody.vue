<template>
  <tab-body
    v-if="active"
    :ref="tab"
    class="tab"
    :active="active"
    :name="tab"
    :aria-selected="active"
  >
    <slot></slot>
  </tab-body>
</template>

<script>
import { mapState } from 'redux-vuex'

import { Tab } from '@podlove/components'

import { tabs } from 'store/selectors'

export default {
  components: {
    TabBody: Tab.Body
  },
  props: {
    tab: {
      type: String,
      default: null
    }
  },
  data: mapState({
    tabs: tabs
  }),
  computed: {
    active() {
      return this.tabs[this.tab]
    }
  }
}
</script>
<style lang="scss">
.tab-content {
  overflow: auto;
  padding: 0 1.5em 1em 1.5em;
}
</style>
