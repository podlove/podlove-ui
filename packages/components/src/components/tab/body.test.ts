import { describe, test, expect } from 'vitest';
import { mount } from '@vue/test-utils'

import TabBody from './Body.vue'

describe('TabBody', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(TabBody, {
        slots: {
          default: '<div>Inner Tab</div>'
        }
      })

      expect(wrapper.vm).toBeTruthy()
    })
  })

  describe('props', () => {
    test('should render without props', async () => {
      const wrapper = await mount(TabBody, {
        slots: {
          default: '<div>Inner Tab</div>'
        }
      })

      expect(wrapper.element).toMatchSnapshot()
    })

    const props = [
      {
        prop: 'active',
        value: true
      },
      {
        prop: 'name',
        value: 'tab-name'
      },
      {
        prop: 'background',
        value: '#fff'
      },
      {
        prop: 'fixed',
        value: true
      }
    ]

    props.forEach(({ prop, value }) => {
      test(`should render with prop ${prop}`, async () => {
        const wrapper = await mount(TabBody, {
          slots: {
            default: '<div>Inner Tab</div>'
          },
          props: {
            [prop]: value
          }
        })

        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
