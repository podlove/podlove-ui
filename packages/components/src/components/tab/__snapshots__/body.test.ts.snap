// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TabBody > props > should render with prop active 1`] = `
<div
  aria-hidden="false"
  aria-labelledby="trigger-"
  class="podlove-component-tab-body overflow-hidden active overflow-y-auto"
  data-v-e2090dd9=""
  role="tabpanel"
  tabindex="0"
>
  <div
    class="tab-content"
    data-v-e2090dd9=""
  >
    
    <div>
      Inner Tab
    </div>
    
  </div>
</div>
`;

exports[`TabBody > props > should render with prop background 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="trigger-"
  background="#fff"
  class="podlove-component-tab-body overflow-hidden"
  data-v-e2090dd9=""
  role="tabpanel"
  tabindex="0"
>
  <div
    class="tab-content"
    data-v-e2090dd9=""
  >
    
    <div>
      Inner Tab
    </div>
    
  </div>
</div>
`;

exports[`TabBody > props > should render with prop fixed 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="trigger-"
  class="podlove-component-tab-body overflow-hidden fixed-height absolute top-0 left-0 w-full overflow-x-hidden"
  data-v-e2090dd9=""
  role="tabpanel"
  tabindex="0"
>
  <div
    class="tab-content"
    data-v-e2090dd9=""
  >
    
    <div>
      Inner Tab
    </div>
    
  </div>
</div>
`;

exports[`TabBody > props > should render with prop name 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="trigger-tab-name"
  class="podlove-component-tab-body overflow-hidden"
  data-v-e2090dd9=""
  role="tabpanel"
  tabindex="0"
>
  <div
    class="tab-content"
    data-v-e2090dd9=""
  >
    
    <div>
      Inner Tab
    </div>
    
  </div>
</div>
`;

exports[`TabBody > props > should render without props 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="trigger-"
  class="podlove-component-tab-body overflow-hidden"
  data-v-e2090dd9=""
  role="tabpanel"
  tabindex="0"
>
  <div
    class="tab-content"
    data-v-e2090dd9=""
  >
    
    <div>
      Inner Tab
    </div>
    
  </div>
</div>
`;
